g1ant-language/desktop-apps-automation2.md

```diff
@@ -0,0 +1,211 @@
+
+```G1ANT
+♥list = element1❚element2❚element3
+```
+
+The elements are separated with an [array separator](../g1ant-language/special-characters/array-separator.md) (use **Ctrl+|** or `❚` toolbar icon to insert it). When you want to retrieve the value of a given element, you use the list variable name followed by the index of a desired element, e.g. if you wanted to get the value of the third element in the `♥list` variable above, you would write `♥list⟦3⟧`. (Here you have another special character, called variable index; to insert it, use **Ctrl+[** or `⟦⟧` toolbar icon.)
+
+As we already mentioned, the mail structure is a kind of a list itself: it has several fields that store the attributes of a message. To read these elements, you use the same indexing method as with lists, but you use field names as indexes. For example, to get the subject of a message, you would write `♥mail⟦subject⟧` — provided that `♥mail` is a variable that refers to a particular element (message) in the list of emails.
+
+The problem with a list of emails like yours is that not only you have to pick one of its elements (an individual message), but also a particular element of this element (subject in case of this exercise). You will have to figure out how to read messages one by one from the list variable and check their subject fields. Fortunately, there is a loop designed especially for cycling through elements in a list.
+
+### Subject: Foreach and If Loops
+
+The `foreach` loop command picks the first element from a list, executes specified actions, then moves to the next element and so forth, until the last element is processed. For the purposes of this exercise you would want to check the message subject for each message stored in the retrieved emails list variable. Let’s start translating it into G1ANT.Language:
+
+```G1ANT
+procedure checksubject
+    foreach ♥message in ♥emails
+```
+
+You are defining the `checksubject` procedure, which begins with the `foreach` loop. This loop will cycle through `♥emails` list variable, reading each of its elements one by one and assigning it to the `♥message` variable.
+
+Now you want to check the subject of the `♥message` currently processed by the `foreach` loop: if the message subject contains word “unsubscribe”, then delete the sender’s email from the mailing list. Guess what? G1ANT.Language provides `if` conditional block command. Its based on C# conditional logic, so most of the times you will need to use the `⊂⊃` characters and provide expressions as C# snippet.
+
+You could write the condition this way:
+
+```G1ANT
+if ⊂♥message⟦subject⟧ == "unsubscribe"⊃
+```
+
+It would work perfectly — unless the message subject contains more characters or words, “unsubscribe me” for instance. The code above checks if the subject exactly matches the given text. If you want to be sure  that any occurence of “unsubscribe” in the subject will make the condition true, use this snippet:
```

Comment:
ToLowerCase() (in case of "Unsubscribe") might be a good idea as well.

`sure [two spaces] that` => sure that

Reply:
```suggestion
It would work perfectly — unless the message subject contains more characters or words, “unsubscribe me” for instance. The code above checks if the subject exactly matches the given text. If you want to be sure that any occurence of “unsubscribe” in the subject will make the condition true, use this snippet:
```

Reply:
Lower case function would be welcome, but I think it's a bit too much for one new thing. And I don't know how to use it in the snippet. :smile: